lib/OpenQA/Utils.pm

```diff
@@ -296,22 +297,40 @@ sub parse_assets_from_settings {
     return $assets;
 }
 
+my %bugrefs = (
+    bnc => 'https://bugzilla.novell.com/show_bug.cgi?id=',
+    bsc => 'https://bugzilla.suse.com/show_bug.cgi?id=',
+    boo => 'https://bugzilla.opensuse.org/show_bug.cgi?id=',
+    poo => 'https://progress.opensuse.org/issues/',
+);
+my %bugurls = reverse %bugrefs;
+
 sub bugurl {
     my ($bugref) = @_;
-    my %bugrefs = (
-        bnc => 'https://bugzilla.novell.com/show_bug.cgi?id=',
-        bsc => 'https://bugzilla.suse.com/show_bug.cgi?id=',
-        boo => 'https://bugzilla.opensuse.org/show_bug.cgi?id=',
-        poo => 'https://progress.opensuse.org/issues/',
-    );
     return $bugrefs{$bugref};
 }
 
 sub bugref_to_href {
     my ($text) = @_;
+    my $regex = join('|', values %bugurls);
+    $regex = qr/(($regex)#(\d+))(\s|$|\n|\r)/;
+    $text =~ s{$regex}{<a href="@{[$bugrefs{$2}]}$3">$1</a>}gi;
+    return $text;
+}
 
-    $text =~ s{((bnc|bsc|boo|poo)#(\d+))}{<a href="@{[bugurl($2)]}$3">$1</a>}gi;
+sub href_to_bugref {
+    my ($text) = @_;
+
+    my @urlregexes;
+    for (values %bugrefs) {
+        # escape ?
+        my ($r) = $_ =~ s/\?/\\\?/gr;
+        push @urlregexes, $r;
+    }
 
+    my $regex = join('|', @urlregexes);
+    $regex = qr/($regex)(\d+)(\s|$|\n|\r)/;
+    $text =~ s{$regex}{@{[$bugurls{$1}]}#$2}gi;
```

Comment:
I think you still need to append $3 at the end. Otherwise you'll consume trailing whitespace. Try using url in text.